connect_ext_test.go

```diff
@@ -521,6 +521,33 @@ func TestHeaderBasic(t *testing.T) {
 	assert.Equal(t, response.Header().Get(key), hval)
 }
 
+func TestHeaderHost(t *testing.T) {
+	t.Parallel()
+	const (
+		key  = "Host"
+		cval = "buf.build"
+	)
+
+	pingServer := &pluggablePingServer{
+		ping: func(_ context.Context, request *connect.Request[pingv1.PingRequest]) (*connect.Response[pingv1.PingResponse], error) {
+			assert.Equal(t, request.Header().Get(key), cval)
+			response := connect.NewResponse(&pingv1.PingResponse{})
+			return response, nil
+		},
+	}
+	mux := http.NewServeMux()
+	mux.Handle(pingv1connect.NewPingServiceHandler(pingServer))
+	server := httptest.NewServer(mux)
+	defer server.Close()
+
+	client := pingv1connect.NewPingServiceClient(server.Client(), server.URL)
```

Comment:
Can you try also testing this flow using the `connect.WithGRPC()` and `connect.WithGRPCWeb()` client options? I suspect that'll change how you approach fixing the bug 😉

Reply:
Thanks for the quick review! I improved the test case and moved the logic. Still not sure if this is the right position for it.